config/routes.rb

```diff
@@ -36,4 +46,10 @@
   get "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinary_offices#veterinarians"
   get "/veterinary_offices/:veterinary_office_id/veterinarians/new", to: "veterinarians#new"
   post "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinarians#create"
-end
+
+  get "/admin/shelters", to: "admin/shelters#index"
+  get "/admin/applications", to: "admin/applications#index"
+  get "/admin/applications/:id", to: "admin/applications#show"
+  patch "/admin/applications/:id", to: "admin/application_pets#update"
```

Comment:
Referring back to the comment I left in your admin/application_pets_controller -- this is not an external route that a user has to go to, but is a route for a data operation to update the application_pets resource. So, it's okay to just put it in the other ApplicationPetsController, and then after updating, you can redirect back to an admin page